demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -712,32 +729,17 @@ static void otaAppCallback( OtaJobEvent_t event,
     {
         LogInfo( ( "Received OtaJobEventActivate callback from OTA Agent." ) );
 
-        if( pthread_mutex_lock( &mqttMutex ) == 0 )
-        {
-            /* OTA job is completed. so delete the network connection. */
-            MQTT_Disconnect( &mqttContext );
-
-            pthread_mutex_unlock( &mqttMutex );
-        }
-        else
-        {
-            LogError( ( "Failed to acquire mutex to execute MQTT_Disconnect"
-                        ",errno=%s",
-                        strerror( errno ) ) );
-        }
-
-        /* Clear the mqtt session flag. */
-        mqttSessionEstablished = false;
+        /* Disconnect from broker and close connection. */
+        disconnect();
 
         /* Activate the new firmware image. */
         OTA_ActivateNewImage();
 
         /* We should never get here as new image activation must reset the device.*/
-        LogError( ( "New image activation failed." ) );
+        LogError( ( "New image activation failed, exiting." ) );
 
-        for( ; ; )
-        {
-        }
+        /* Exit this application.*/
+        exit( EXIT_SUCCESS );
```

Comment:
If we failed to activate the image, EXIT_SUCCESS seems to not make sense. Is there a different error code that we can use?

Reply:
At this point the device has successfully downloaded the new version, activation only failed as user will have to execute the application manually. So this is not a error , we just log that activation failed.